src/main/java/software/sigma/bu003/internship/vedmid_andrii/service/PartService.java

```diff
@@ -0,0 +1,75 @@
+package software.sigma.bu003.internship.vedmid_andrii.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import software.sigma.bu003.internship.vedmid_andrii.entity.Part;
+import software.sigma.bu003.internship.vedmid_andrii.exception.PartNotFoundException;
+import software.sigma.bu003.internship.vedmid_andrii.repository.PartRepository;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+@AllArgsConstructor
+public class PartService {
+
+    private final PartRepository partRepository;
+
+    public Part createPart(Part part) {
+        checkIsPart(part.getBrand(), part.getCode());
+
+        Part partFromDB = partRepository.findByBrandAndCode(part.getBrand(), part.getCode());
+
+        if(!Objects.isNull(partFromDB)){
+            return partFromDB;
```

Comment:
avoid using multiple return, also what will be in Mongo if you try to create the same record? Do you really need to check if this record is already there?